### PR TITLE
Fix docker multiarch build for operator

### LINKED ISSUE #874: Multiarch support seems to be incomplete
Hi,

I tried to install the opentelemetry-operator via helm chart on a microk8s running on Apple M1:
```
helm install \
  opentelemetry-operator open-telemetry/opentelemetry-operator -n  opentelemetry-operator-system \
  --set manager.image.tag=v0.50.0
```

According to github the image used should be multiarch:
https://github.com/open-telemetry/opentelemetry-operator/pkgs/container/opentelemetry-operator%2Fopentelemetry-operator/21667125?tag=v0.50.0

Immediately after starting the pod crashes with `standard_init_linux.go:228: exec user process caused: exec format error`.

Do you plan to support the M1 architecture?

Thanks for your help!
Cheers

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Resolves #874

### CODE REVIEW

**Review by devops-42 — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -17,7 +17,6 @@ COPY internal/ internal/
 COPY pkg/ pkg/
 COPY versions.txt versions.txt
 
-ARG TARGETARCH=amd64
```
Comment (frzifus) [inline Dockerfile:20]:
```suggestion
ARG TARGETARCH
```
would it make sense to remove `=amd64`?

Comment (pavolloffay) [reply Dockerfile:20]: Then we would have to pass the arch as argument which should not be necessary

**Review by VineethReddy02 — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (frzifus) [reply Dockerfile:20]: But without we are not able to build it for a different target architecture, right?

Comment (pavolloffay) [reply Dockerfile:20]: we are via docker buildx

**Pull request merged**